admin/class-admin-apple-meta-boxes.php

```diff
@@ -386,26 +305,84 @@ public static function section_is_checked( $sections, $section_id, $is_default )
 	/**
 	 * Registers assets used by meta boxes.
 	 *
-	 * @param string $hook
+	 * @param string $hook The initiator of the action hook.
+	 *
 	 * @access public
 	 */
 	public function register_assets( $hook ) {
+
+		// Only fire on post and new post views.
 		if ( 'post.php' !== $hook && 'post-new.php' !== $hook ) {
 			return;
 		}
 
-		wp_enqueue_style( $this->plugin_slug . '_meta_boxes_css', plugin_dir_url(
-			__FILE__ ) .  '../assets/css/meta-boxes.css' );
+		// Ensure media modal assets are enqueued.
+		wp_enqueue_media();
 
-		wp_enqueue_script( $this->plugin_slug . '_meta_boxes_js', plugin_dir_url(
-			__FILE__ ) .  '../assets/js/meta-boxes.js', array( 'jquery' ),
-			self::$version, true );
+		// Enqueue metabox stylesheet.
+		wp_enqueue_style(
```

Comment:
I don't like how the enqueues and localization have to happen in two different files for the same scripts. It seems like a maintenance disaster waiting to happen later on. Since these both have the same parent class, can we leverage that to only do it once?